src/main/java/kotlinx/reflect/lite/builtins/JvmPrimitiveType.kt

```diff
@@ -0,0 +1,30 @@
+package kotlinx.reflect.lite.builtins
+
+import kotlinx.reflect.lite.name.*
+
+enum class JvmPrimitiveType(val primitiveType: PrimitiveType, val javaKeywordName: String, wrapperClassName: String) {
+    BOOLEAN(PrimitiveType.BOOLEAN, "boolean", "Boolean"),
+    CHAR(PrimitiveType.CHAR, "char", "Character"),
+    BYTE(PrimitiveType.BYTE, "byte", "Byte"),
+    SHORT(PrimitiveType.SHORT, "short", "Short"),
+    INT(PrimitiveType.INT, "int", "Integer"),
+    FLOAT(PrimitiveType.FLOAT, "float", "Float"),
+    LONG(PrimitiveType.LONG, "long", "Long"),
+    DOUBLE(PrimitiveType.DOUBLE, "double", "Double");
+
+    val wrapperFqName = FqName("java.lang.$wrapperClassName")
+
+    companion object {
+        private val TYPE_BY_NAME: MutableMap<String, JvmPrimitiveType> =
+            HashMap()
+
+        operator fun get(name: String): JvmPrimitiveType =
+            TYPE_BY_NAME[name] ?: throw AssertionError("Non-primitive type name passed: $name")
+
+        init {
+            for (type in values()) {
+                TYPE_BY_NAME[type.javaKeywordName] = type
+            }
+        }
```

Comment:
Style:

Github Copilot impressed me a lot just now by suggesting this code (cc @qwwdfsad):
```suggestion
        private val TYPE_BY_NAME: Map<String, JvmPrimitiveType> =
            values().associateBy { it.javaKeywordName }

        operator fun get(name: String): JvmPrimitiveType =
            TYPE_BY_NAME[name] ?: throw AssertionError("Non-primitive type name passed: $name")
```